services/motion/builtin/move_request.go

```diff
@@ -137,20 +137,25 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 	}
 	localizer := motion.NewMovementSensorLocalizer(movementSensor, origin, movementSensorToBase.Pose())
 
-	// convert destination into spatialmath.Pose with respect to where the localizer was initialized
-	goal := spatialmath.GeoPointToPose(destination, origin)
-
-	// convert GeoObstacles into GeometriesInFrame with respect to the base's starting point
-	geoms := spatialmath.GeoObstaclesToGeometries(obstacles, origin)
+	// create a KinematicBase from the componentName
+	baseComponent, ok := ms.components[componentName]
+	if !ok {
+		return nil, resource.NewNotFoundError(componentName)
+	}
+	b, ok := baseComponent.(base.Base)
+	if !ok {
+		return nil, fmt.Errorf("cannot move component of type %T because it is not a Base", baseComponent)
+	}
 
-	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
-	worldState, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	fs, err := ms.fsService.FrameSystem(ctx, nil)
 	if err != nil {
 		return nil, err
 	}
 
+	// Important: this assumes a 0 degree heading. This, and the geoms below, need to be transformed to the correct pose.
+	goalPoseRaw := spatialmath.GeoPointToPose(destination, origin)
```

Comment:
[ulta-opt]: could you add the why? This might be hard to figure out if a user chooses to read source code.

Reply:
I will always ultra-opt for more, clearer documentation :)

Reply:
How do you like it now?

Reply:
This is perfect. Thank you!